proxy/tracing.go

```diff
@@ -112,26 +110,34 @@ func (t *proxyTracing) startFilterTracing(operation string, ctx *context) *filte
 	if t.disableFilterSpans {
 		return nil
 	}
-	span := tracing.CreateSpan(operation, ctx.request.Context(), t.tracer)
+
+	span, reqCtx := t.StartSpan(ctx.request.Context(), operation)
+	ctx.request = ctx.request.WithContext(reqCtx)
 	ctx.parentSpan = span
 
 	return &filterTracing{span, t.logFilterLifecycleEvents}
 }
 
 func (t *filterTracing) finish() {
-	if t != nil {
-		t.span.Finish()
+	if t == nil {
```

Comment:
minor: why did you decide to change those functions at all? It looks it's still better to leave the as it is:

+ the code before and after have the same results
+ the size of this PR will be decreased -> it will be slightly easier to read

Reply:
Because if I don't change it will expect a open tracing tracer interface and we no longer have access to this tracer in this part of the code, also the idea is to move away from open tracing so keep this interface here would not be productive for how we want this package to evolve.